### PR TITLE
Prepare two governors

### PR DESCRIPTION
**Pull request by havok1231**

### CODE REVIEW

**Comment by havok1231** [comment]
I have some questions about this task but you can see what has already done.

File: tests-js/Governance/loan-token/LoanTokenTest.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+const PreviousLoanToken = artifacts.require('PreviousLoanToken');
+
+const TOTAL_SUPPLY = 100;
+
+const DAY = 86400;
+const TWO_DAYS = 86400 * 2;
+const TWO_WEEKS = 86400 * 14;
+const MAX_DURATION = new BN(24 * 60 * 60).mul(new BN(1092));
+
+contract('GovernanceIntegration', accounts => {
```
Comment (jameshowlett977) [inline tests-js/Governance/loan-token/LoanTokenTest.js:26]: LoanTokenUpgrade ?

File: tests-js/Governance/loan-token/LoanTokenTest.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+            //check that previous admin is address(0)
+            let admin = await loanToken.admin();
+            assert.equal(admin, constants.ZERO_ADDRESS);
+
+            //change admin
+            loanToken.init(token.address, "SomeName", "SomeSymbol", root);
+
+            admin = await loanToken.admin();
+            assert.equal(admin, root)
```
Comment (jameshowlett977) [inline tests-js/Governance/loan-token/LoanTokenTest.js:85]: Please add reading of a few old fields.
Also check modifier `onlyAdmin` before and after upgrade.

File: tests-js/Governance/loan-token/LoanTokenTest.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        protocolSettings = await ProtocolSettings.new();
+        protocol = await Protocol.new();
+        await protocol.replaceContract(protocolSettings.address);
+        protocol = await ProtocolSettings.at(protocol.address);
+        await protocol.transferOwnership(timelock.address);
+    });
+
+    describe("change settings", () => {
+        //@todo rename this or delete, idk why I wrote this test
+        it("check how works proxy", async () => {
```
Comment (jameshowlett977) [inline tests-js/Governance/loan-token/LoanTokenTest.js:68]: I propose to add the same test for `ProtocolSettings` upgrade.

**Review by jameshowlett977 — COMMENTED** [review]
LoanTokenSettings was refactored in this PR https://github.com/DistributedCollective/Sovryn-smart-contracts/pull/71
You need to merge version from development to your branch.

File: contracts/connectors/loantoken/LoanToken.sol [hunk 4]
```diff
@@ -7,13 +7,14 @@ pragma solidity 0.5.17;
 
 import "./AdvancedTokenStorage.sol";
 
-
+//@todo can I change this proxy to EIP-1822 proxy standard, please. https://eips.ethereum.org/EIPS/eip-1822. It's really hard to work with this.
```
[the remark this answers is not in the document]
Comment (havok1231) [reply contracts/connectors/loantoken/LoanToken.sol:10]: +

File: contracts/connectors/loantoken/LoanToken.sol [hunk 5]
```diff
@@ -80,11 +81,13 @@ contract LoanToken is AdvancedTokenStorage {
...
     }
-    
+
+    //@todo add check for double init, idk but init usually can be called only once.
     function initialize(
         address _loanTokenAddress,
         string memory _name,
-        string memory _symbol)
+        string memory _symbol,
+        address _admin)
```
[the remark this answers is not in the document]
Comment (havok1231) [reply contracts/connectors/loantoken/LoanToken.sol:90]: +

File: contracts/connectors/loantoken/LoanTokenSettingsLowerAdmin.sol [hunk 6]
```diff
@@ -9,21 +9,47 @@ pragma experimental ABIEncoderV2;
...
+            msg.sender == admin, "unauthorized");
         _;
     }
 
+    //@todo add check for double init, idk but init usually can be called only once.
+    function init(
+        address _loanTokenAddress,
+        string memory _name,
+        string memory _symbol,
+        address _admin)
```
[the remark this answers is not in the document]
Comment (havok1231) [reply contracts/connectors/loantoken/LoanTokenSettingsLowerAdmin.sol:39]: we don't need 2 same functions in proxy and implementation

[the remark this answers is not in the document]
Comment (jameshowlett977) [reply contracts/connectors/loantoken/LoanTokenSettingsLowerAdmin.sol:39]: It's not the same function, it's the same comment :-)

File: contracts/connectors/loantoken/LoanToken.sol [hunk 7]
```diff
@@ -7,13 +7,14 @@ pragma solidity 0.5.17;
...
 
-
+//@todo can I change this proxy to EIP-1822 proxy standard, please. https://eips.ethereum.org/EIPS/eip-1822. It's really hard to work with this.
 contract LoanToken is AdvancedTokenStorage {
 
     // It is important to maintain the variables order so the delegate calls can access sovrynContractAddress and wrbtcTokenAddress
     address public sovrynContractAddress;
     address public wrbtcTokenAddress;
     address internal target_;
+    address public admin;
```
[the remark this answers is not in the document]
Comment (havok1231) [reply contracts/connectors/loantoken/LoanToken.sol:17]: Yes, I've thought that LoanToken -> LoanTokenSettingsLowerAdmin, as you wrote in task, but now it's LoanToken  -> LoanTokenLogicStandard

File: tests-js/Governance/loan-token/LoanTokenTest.js (hunk 1 above)
Comment (havok1231) [reply tests-js/Governance/loan-token/LoanTokenTest.js:26]: +

File: contracts/connectors/loantoken/LoanTokenSettingsLowerAdmin.sol (hunk 6 above)
[the remark this answers is not in the document]
Comment (havok1231) [reply contracts/connectors/loantoken/LoanTokenSettingsLowerAdmin.sol]: +

File: contracts/connectors/loantoken/LoanToken.sol (hunk 7 above)
[the remark this answers is not in the document]
Comment (havok1231) [reply contracts/connectors/loantoken/LoanToken.sol:17]: +

File: tests-js/Governance/loan-token/LoanTokenTest.js (hunk 2 above)
Comment (havok1231) [reply tests-js/Governance/loan-token/LoanTokenTest.js:90]: +

**Review by ororopickpocket — APPROVED** [review]

**Pull request merged**